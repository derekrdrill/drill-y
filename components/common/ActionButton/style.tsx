import styled from 'styled-components';
import { Button } from '@mui/material';

export const ActionButtonRoot = styled(Button)<{
  $borderRadius?: number;
  $buttonColor?: string;
  $buttonHoverColor?: string;
  $buttonWidth?: string | number;
  $marginTop?: number;
  disabled?: boolean;
  variant?: string;
}>(
  ({
    $borderRadius,
    $buttonColor,
    $buttonHoverColor,
    $buttonWidth,
    $marginTop,
    disabled,
    variant,
  }) => ({
    '&:hover': {
      backgroundColor: `${$buttonHoverColor} !important`,
      color: `#FFFFFF !important`,
    },
    backgroundColor: `${variant === 'contained' ? `${$buttonColor}` : 'inherit'} !important`,
    borderColor: disabled
      ? 'gainsboro'
      : `${variant === 'outlined' ? `${$buttonColor}` : 'inherit'} !important`,
    borderRadius: `${$borderRadius ?? 75}px !important`,
    boxShadow: 'none !important',
    color: disabled
      ? 'gainsboro'
      : `${variant === 'outlined' ? `${$buttonColor}` : '#FFFFFF'} !important`,
    fontFamily: `'Josefin Sans', sans-serif !important`,
    fontWeight: '600 !important',
    fontSize: '12px !important',
    marginTop: `${$marginTop ?? 4}px !important`,
    paddingBottom: '4px !important',
    paddingLeft: '20px !important',
    paddingRight: '20px !important',
    paddingTop: '6px !important',
    width: $buttonWidth
      ? `${typeof $buttonWidth === `number` ? `${$buttonWidth}px` : $buttonWidth} !important`
      : 'inherit',
  }),
);
