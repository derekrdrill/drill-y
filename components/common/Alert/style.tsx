import styled from 'styled-components';
import { Grid } from '@mui/material';

// export const AlertContainer = styled(Grid)<{ isAlertHidden?: boolean; isAlertFading?: boolean }>(
//   ({ isAlertHidden, isAlertFading }) => ({
//     display: isAlertHidden && `none !important`,
//     opacity: isAlertHidden || isAlertFading ? 0 : 1,
//     position: 'fixed',
//     top: 125,
//     transition: 'all 350ms ease-in',
//     zIndex: 6,
//   }),
// );

export const AlertContainer = styled(Grid)<{
  $isAlertHidden?: boolean | undefined;
  $isAlertFading?: boolean | undefined;
}>(({ $isAlertHidden, $isAlertFading }) => ({
  display: `${$isAlertHidden ? 'none' : 'inline-flex'} !important`,
  opacity: $isAlertHidden || $isAlertFading ? 0 : 1,
  position: 'fixed',
  top: 125,
  transition: 'all 350ms ease-in',
  zIndex: 6,
}));
