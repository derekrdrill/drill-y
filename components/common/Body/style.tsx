import styled from 'styled-components';
import { Grid } from '@mui/material';

export const BodyRoot = styled(Grid)({
  zIndex: 0,
});
