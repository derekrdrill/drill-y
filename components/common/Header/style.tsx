import styled from 'styled-components';
import { AppBar, Grid } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const HeaderRoot = styled(AppBar)({
  backgroundColor: 'white !important',
  borderBottom: '1px solid #DFDFDF',
  boxShadow: 'none !important',
  zIndex: '3 !important',
});

export const HeaderTop = styled(Grid)({
  borderBottom: '1px solid #DFDFDF',
  padding: '5px 0',
});

export const HeaderLogo = styled.img({
  '&:hover': {
    opacity: 0.8,
  },
  cursor: 'pointer',
  height: 70,
  width: 100,
});

export const HeaderMenuText = styled.h4({
  '&:hover': {
    backgroundPositionX: '0%',
    backgroundSize: '100% 0.1em',
  },
  backgroundImage: 'linear-gradient(#212121, #212121)',
  backgroundPositionX: '100%',
  backgroundPositionY: '100%',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '0% 0.1em',
  color: 'black',
  cursor: 'pointer',
  fontFamily: `'Josefin Sans', sans-serif !important`,
  fontWeight: '600 !important',
  textDecoration: 'none',
  textTransform: 'uppercase',
  transition: 'background-size 0.2s ease-in-out',
});

export const HeaderSearchIcon = styled(SearchIcon)({
  '&:hover': {
    color: '#a501c6',
    transform: 'scale(1.6)',
  },
  cursor: 'pointer',
  marginTop: 7,
  transition: 'all 5s ease-in-out',
  transform: 'scale(1.4)',
});
