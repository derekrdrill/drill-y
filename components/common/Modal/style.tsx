import styled from 'styled-components';
import { Grid } from '@mui/material';

export const ModalRootContainer = styled(Grid)({
  position: 'absolute',
  top: 200,
});

export const ModalContainer = styled(Grid)({
  backgroundColor: '#FFFFFF !important',
  borderRadius: 7,
  padding: '10px 20px',
});

export const ModalRow = styled(Grid)<{ isBottom?: boolean; isBody?: boolean }>(
  ({ isBottom, isBody }) => ({
    borderBottom: !isBottom ? '1px gainsboro solid' : 'none',
    margin: !isBottom ? '10px 0' : 0,
    padding: isBody ? '20px 0' : '5px 0',
  }),
);
