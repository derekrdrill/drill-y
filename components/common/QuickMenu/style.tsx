import styled from 'styled-components';
import { Grid } from '@mui/material';

export const QuickMenuRoot = styled(Grid)<{ $isQuickMenuOpen?: boolean }>(
  ({ $isQuickMenuOpen }) => ({
    backgroundColor: '#FFFFFF',
    boxShadow: '0 4px 2px -2px gray',
    left: 0,
    height: 400,
    minWidth: '100%',
    position: 'fixed',
    top: 123,
    transform: $isQuickMenuOpen ? 'none' : 'translateY(-105%)',
    transition: 'transform 300ms linear',
    width: '100%',
    zIndex: 2,
  }),
);
