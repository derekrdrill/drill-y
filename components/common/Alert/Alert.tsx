import * as React from 'react';
import { Alert as AlertComponent, Grid } from '@mui/material';

import GlobalContext from '../../../context/GlobalContext';

import { AlertContainer } from './style';
import { setAlertItem } from './actions/AlertActions';

const Alert = () => {
  const {
    dispatch,
    state: { alertItem },
  } = React.useContext(GlobalContext);

  const [isAlertFading, setIsAlertFading] = React.useState<boolean>(false);

  React.useEffect(() => {
    if (alertItem?.isAlertOpen) {
      setTimeout(() => {
        setIsAlertFading(true);
      }, 3000);

      setTimeout(async () => {
        setIsAlertFading(false);
        dispatch(await setAlertItem('', false));
      }, 3200);
    }
  }, [alertItem?.isAlertOpen]);

  return (
    <AlertContainer
      container
      $isAlertHidden={!alertItem?.isAlertOpen}
      $isAlertFading={isAlertFading}
    >
      <Grid item xs={1} sm={2} md={4} />
      <Grid item xs={10} sm={8} md={4}>
        <AlertComponent
          onClose={
            /* istanbul ignore next */
            async () => {
              dispatch(await setAlertItem(alertItem?.alertMessage, false));
            }
          }
          severity={alertItem?.alertSeverity}
        >
          {alertItem?.alertMessage}
        </AlertComponent>
      </Grid>
      <Grid item xs={1} sm={2} md={4} />
    </AlertContainer>
  );
};

export default Alert;
