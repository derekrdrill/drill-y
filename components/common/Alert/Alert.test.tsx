import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Alert from './Alert';

import { AlertContainer } from './style';

describe('Alert unit tests', () => {
  it('renders as expected', () => {
    const result = renderer.create(<Alert />).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders AlertContainer correctly (isAlertHidden)', async () => {
    render(<AlertContainer data-testid='alert-container' isAlertHidden />);

    const alertContainer = await screen.findAllByTestId('alert-container');

    expect(alertContainer[0]).toHaveClass('sc-beqWaB czspmq');
  });

  it('renders AlertContainer correctly (!isAlertHidden)', async () => {
    render(<AlertContainer data-testid='alert-container' />);

    const alertContainer = await screen.findAllByTestId('alert-container');

    expect(alertContainer[0]).toHaveClass('sc-beqWaB aIxVR');
  });
});
