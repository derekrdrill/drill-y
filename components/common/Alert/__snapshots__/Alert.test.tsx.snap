// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container sc-beqWaB czspmq css-11lq3yg-MuiGrid-root"
  isAlertFading={false}
  isAlertHidden={true}
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-2 MuiGrid-grid-md-4 css-y734yy-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-8 MuiGrid-grid-md-4 css-llikqt-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="SuccessOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      />
      <div
        className="MuiAlert-action css-ki1hdl-MuiAlert-action"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Close"
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-2 MuiGrid-grid-md-4 css-y734yy-MuiGrid-root"
  />
</div>
`;
